crates/core/src/commands/check.rs

```diff
@@ -37,7 +37,54 @@ pub struct CheckOptions {
     pub read_data: bool,
 }
 
-impl CheckOptions {
+#[derive(Debug, PartialEq, Eq)]
+#[non_exhaustive]
+/// `CheckErrorLevel` describes severity levels of problems identified by check.
+pub enum CheckErrorLevel {
+    /// A warning: Something is strange and should be corrected, but repository integrity is not affected.
+    Warn,
+    /// An erro: Something in the repository is not as it should be.
+    Error,
+}
+
+#[derive(Debug)]
+/// `CheckResults` is a list of errors encountered during the check.
+pub struct CheckResults {
+    /// The list of errors with severity level.
+    pub errors: Vec<(CheckErrorLevel, CheckError)>,
+}
```

Comment:
Wondering if we could replace these with: https://crates.io/crates/lazy_errors

Didn't use it, but it sounded interesting on Reddit last week: https://www.reddit.com/r/rust/comments/1ddi1nb/i_published_my_first_crate_lazy_errors/

Reply:
That looks still relevant to me, lazy_errors looks actually really good for collecting multiple errors like CheckResults does but looks more ergonomic to me at first look.